### PR TITLE
Initial Development

### PR DESCRIPTION
**Pull request by EmilyMatt**
## Summary by CodeRabbit

- **New Features**
  - Introduced GitHub Actions workflow for mandatory pull request checks.
  - Added comprehensive README with usage instructions and code examples for FalkorDB Rust client.
  - Implemented a Builder pattern for creating and customizing FalkorDB clients.
  - Added support for synchronous and asynchronous client interfaces.
  - Introduced functionality for handling graph operations, queries, and constraints in FalkorDB.

- **Documentation**
  - Updated `.gitignore` to include and refine specific entries.
  - Added detailed LICENSE file with Server Side Public License Version 1 and Elastic License 2.0.

- **Configuration**
  - Introduced configuration files for Clippy, Deny, and Rustfmt to ensure code quality and formatting standards.

- **Enhancements**
  - Improved connection handling with connection pooling for synchronous clients.
  - Added support for executing commands and managing Redis connections.

- **Bug Fixes**
  - Enhanced error handling and parsing functionalities within the FalkorDB client.

### CODE REVIEW

File: src/client/blocking.rs [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+                .as_i64()
+                .unwrap(),
+            desired_val
+        );
+
+        client
+            .config_set("DELTA_MAX_PENDING_CHANGES", current_val)
+            .ok();
+    }
+}
```
[the remark this answers is not in the document]
Comment (EmilyMatt) [reply src/client/blocking.rs]: Yes, please

**Review by AviAvni — APPROVED** [review]

File: src/response/index.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        let val = val.into_vec()?;
+        let mut field_types = Vec::with_capacity(val.len());
+        for field_type in val {
+            field_types.push(field_type.into_string()?.as_str().try_into()?);
+        }
+
+        out_map.insert(key, field_types);
+    }
+
+    Ok(out_map)
```
[the remark this answers is not in the document]
Comment (EmilyMatt) [reply src/response/index.rs:50]: The ? Operator cannot be used within closures, so this is not viable.
I could use filter_map, but that would defeat the function of error propegating

File: src/client/blocking.rs [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+            .flat_map(|config| {
+                let [key, val]: [FalkorValue; 2] = config
+                    .into_vec()?
+                    .try_into()
+                    .map_err(|_| FalkorDBError::ParsingArrayToStructElementCount)?;
+
+                Result::<_, FalkorDBError>::Ok((key.into_string()?, ConfigValue::try_from(val)?))
+            })
+            .collect::<HashMap<String, ConfigValue>>())
+    }
```
[the remark this answers is not in the document]
Comment (EmilyMatt) [reply src/client/blocking.rs]: The ? operator which propagates errors upward cannot be used in closures

File: src/parser/mod.rs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+use crate::{FalkorResult, FalkorValue, GraphSchema};
+
+/// This trait allows implementing a parser from the table-style result sent by the database, to any other struct
+pub trait FalkorParsable: Sized {
+    /// Parse the following value, using the graph schem owned by the graph object, and the connection used to make the request
+    fn from_falkor_value(
+        value: FalkorValue,
+        graph_schema: &mut GraphSchema,
+    ) -> FalkorResult<Self>;
```
[the remark this answers is not in the document]
Comment (EmilyMatt) [reply src/parser/mod.rs:16]: This is a trait, there is no implementation here

**Review by AviAvni — APPROVED** [review]

**Pull request merged**